### DISCUSSION TITLE
View AWS Cost for the month

### DISCUSSION THREAD

**Discussion by lannonbr** [johnlindquist/kit#73] [Share]
I wrote a script this morning that uses the `aws-cli` to check the cost of my AWS account for a particular time-range:

```js
let start = await arg('Enter start date (YYYY-MM-DD)')
let end = await arg('Enter end date (YYYY-MM-DD)')

let {stdout} = exec(`aws ce get-cost-and-usage --time-period Start=${start},End=${end} --granularity MONTHLY --metrics "BlendedCost"`)

let data = JSON.parse(stdout)

let monthCost = data.ResultsByTime[0].Total.BlendedCost

let { say } = await kit('speech')
await say(`Your AWS bill is $${Number(monthCost.Amount).toFixed(2)}`)
```

I'm thinking in the future of expanding this so if I can get other time granularities (weekly, daily), or drill down based on costs of items based on a tag.

**Comment by johnlindquist** [comment]
Awesome, Benjamin! Thanks so much for sharing!

One of the main purposes of Script Kit is serving as a UI wrapper around CLIs and this encapsulates that idea perfectly.

Have you tried a  [natural language date parser](https://www.npmjs.com/package/chrono-node)? I've been considering adding a date picker of some sort but wondering if it's unnecessary. 🤔

**Reply by lannonbr** [reply]
I haven't yet, looks cool, but as well I am thinking on top of entering start / end times having presets like "Enter a month", "this month", "this year", etc.

Also, I am digging into the source for Script Kit and looking at the `arg` and `prompt` functions and I am interested in using the validate option so if I am looking for a date string, I can enforce a particular syntax for the date format

**Reply by johnlindquist** [reply]
`validate` must return `true` to pass. Return a string if it's invalid. 
(I need to put in logic to handle `false`)

```js
let value = await arg({
  message: 'Type "hi"',
  validate: input =>
    input === "hi" || 'You _MUST_ type "hi"',
})

console.log(`${value} can only be "hi"`)
```

Note: invoking a script from the terminal uses [enquirer](https://github.com/enquirer/enquirer). So I need to stay in-line with their validation approach (or eventually fork enquirer).
